src/templateengine.vala

```diff
@@ -84,8 +84,35 @@ namespace Valdo.TemplateEngine {
                 continue;   // don't copy over template.json
 
             if (file_type == FileType.DIRECTORY) {
-                // create an empty directory
-                DirUtils.create_with_parents ((!) project_child.get_path (), 0755);
+                // Test if Directroy name is a template
+                var folder_name = (!)project_child.get_basename ();
+                if (folder_name.has_prefix ("${")) {
```

Comment:
Perhaps this should be able to replace a variable anywhere in the folder name, not just at the beginning?